Improved: Added support to pass productStore ShopId while calling Shopify-related APIs (#37)

R-Sourabh opened the pull request:
### Related Issues
<!--  Put related issue number which this PR is closing. For example #123 -->

#37 

### Short Description and Why It's Useful
<!-- Describe in a few words what is this Pull Request changing and why it's useful -->
- Added support to pass productStore ShopId while calling Shopify-related APIs
- Added a utility function for sorting the list alphabetically for Department, Sales Channel, and Inventory Variance pages.
- Added a utility function for filtering the API response based on expired results.
- Fixed the ion-header border issue.

### Screenshots of Visual Changes before/after (If There Are Any)
<!-- If you made any changes in the UI layer, please provide before/after screenshots -->


### Contribution and Currently Important Rules Acceptance
<!-- Please get familiar with following info -->

- [x] I read and followed [contribution rules](https://github.com/hotwax/company#contribution-guideline)

R-Sourabh committed:
Fixed: issue of shadow in the ion-header by updating the lock file(#37)

R-Sourabh committed:
Improved: Added support to pass `productStoreShopId` while calling Shopify-related APIs (#37).

- Added a utility function for sorting the list alphabetically for Department, Sales Channel, and Inventory Variance pages.
- Added a utility function for filtering the API response based on expired results.

R-Sourabh committed:
Improved: Updated the logic of the function that saves discount values (#37)

ymaheshwari1 commented on src/components/ProductStoreModal.vue:
```diff
@@ -92,6 +93,10 @@ async function updateSubsidiaryId() {
         subsidiaryId: subsidiaryId.value
       });
+      // fetching the shopify shop for the selected product store
+      if(!initialNetSuiteProductStore || netSuiteProductStore.value.productStoreId !== selectedProductStoreId.value) {
```
```suggestion
      if(!initialNetSuiteProductStore || initialNetSuiteProductStore !== selectedProductStoreId.value) {
```

ymaheshwari1 commented on src/utils/index.ts:
```diff
@@ -21,4 +21,12 @@ const generateInternalId = (name: string) => {
+  return data.filter((item: any) => !item[fieldName] || item[fieldName] > Date.now());
+}
+
+const sortByProperty = (array: any, property: any) => {
```
```suggestion
const sortByProperty = (data: any, property: string) => {
```

ymaheshwari1 commented on src/utils/index.ts:
```diff
@@ -21,4 +21,12 @@ const generateInternalId = (name: string) => {
+}
+
+const sortByProperty = (array: any, property: any) => {
+  return array.sort((a: any, b: any) => a[property].toLowerCase().localeCompare(b[property].toLowerCase()));
```
Add handling if the property is not available on the `a` or `b` parameter.

R-Sourabh committed:
Improved: Changed the parameter name and its data type in the sortByProperty utility function (#37)

- Changed the if check condition for fetching the product store Shopify shop.

R-Sourabh committed:
Improved: updated version of your sortByProperty function with added handling for missing properties(#37)

R-Sourabh committed:
Improved: removed the unused ionicon import(#37)

R-Sourabh committed:
Merge branch 'main' into #37-NetSuite-Integration-UI

R-Sourabh committed:
Implemented: Replaced the alert box input for adding NetSuite ID with a direct input field for shipment methods (#37)

- Updated the editNetSuiteId composable to handle input values for editing and updating appropriately.
- Added logic to filter out duplicate shipment methods on the list page.

R-Sourabh committed:
Improved: added support to show the shipment methods product store name on shipment method page(#37)

- Changed the Base price radio value to Base Price (MSRP).
- Updated the locale text file.
- Handled various cases on the Shipment Method page when inputting the NetSuite ID.

R-Sourabh committed:
Implemented:  added deduplicateByField utility function to filter the duplicate api results(#37)

R-Sourabh committed:
Implemented: added support to update NetSuite ID using input fields instead of alert boxes across all pages (#37)

- Refactored some code.
- Added CSS for the input, chip, and NetSuite ID button.
- Added handling for various edge cases.

R-Sourabh committed:
Implemented: Added the 'Facilities' page & its functionality(#37)

R-Sourabh committed:
Improved: remove the check and keep it simple function call (#37)
